stylix/fonts.nix

```diff
@@ -25,38 +25,38 @@ in {
   options.stylix.fonts = {
     serif = mkOption {
       description = mdDoc "Serif font.";
-      type = fontType;
-      default = fromOs [ "fonts" "serif" ] {
+      type = types.nonEmptyListOf fontType;
```

Comment:
Could we make this backwards compatible by automatically converting a single `fontType` into a list? Something like `coercedTo fontType (f: [ f ]) (nonEmptyListOf fontType)`

Reply:
I have no idea what that is, but I'm willing to learn and improve this c:

Reply:
Basically, if the user sets something of the first type (`fontType`) then it uses the function (`f: [ f ]`) to convert it to the second type (`nonEmptyListOf fontType`). It also allows the user to set the second type directly. In this case all the function does is place the single value into a list, so that definitions from before this update will still work.

Reply:
We sadly cannot do this as `fontType` is a submodule and `coercedTo` doesn't allow submodules :c. Do you have another idea?

Reply:
I'm not sure. Perhaps there is something in the style of `mkRemovedOptionModule` which we can use?

Reply:
I'll check that out tonight

Reply:
I might be able to hack `mkMergedOptionModule` to do what we want.

Reply:
Or not. It requires us to name the new module a different thing.

This can be done with `mkChangedOptionModule`, then have `mkChangedOptionModule ["stylix" "fonts" "monospace"] ["stylix" "fonts" "monospaceWithFallback"] updateFontToFontWithFallbackFunction` which will execute the migration.

I prefer the breaking change to this. What do you think?

Reply:
I feel like it was pure luck, but I managed to make a `coercedTo` function that worked for us. It builds and `flake check` passes so it must not be horribly broken? https://github.com/jalil-salame/NixOsConfig/actions/runs/7519600496/job/20468327186